external/pmrrr/include/pmrrr/lapack/odnst.hpp

```diff
@@ -0,0 +1,177 @@
+/**
+	C++ template version of LAPACK routine dlanst.
+	Based on C code translated by f2c (version 20061008).
+*/
+
+#ifndef __ODNST_HPP__
+#define __ODNST_HPP__
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <float.h>
+#include <assert.h>
+
+#include <pmrrr/lapack/odssq.hpp>
+#include <pmrrr/lapack/lapack.hpp>
+
+namespace pmrrr { namespace lapack {
+
+	template<typename FloatingType>
```

Comment:
In an ideal world, it would be part of the main library and use proper C++ functions instead of f2c, but the latter may take a substantial amount of work. You may want to look at how I handled similar issues in ElSuiteSparse in `external/suite_sparse`.